### PR TITLE
PRO-1542 Fix Pool Activity and Tranding History Screen

### PR DESCRIPTION
**Pull request by lalitkumarrock**
https://linear.app/pillarproject/issue/PRO-1542/pillar-wallet-fix-pool-activity-and-trending-history-screen

## Description

- Fixed Tranding History and Pool Activity screen data.

## How Has This Been Tested?

- Tested on Simulator 14 PRO
- Tested on Android Device

## Screenshots (if appropriate):
https://github.com/pillarwallet/pillarwallet/assets/82652040/27f496d1-b317-4d13-a61e-5932a47e7e90

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+
+// Selectors
+import { useRatesPerChain, useFiatCurrency, useSupportedAssetsPerChain } from 'selectors';
+import { useSupportedChains } from 'selectors/chains';
+
+// Models
+import type { AssetDataNavigationParam } from 'models/Asset';
+import type { TredingActivityData, PoolActivityData } from 'models/Exchange';
+
+interface Props {
```
Comment (MayukhSamadder2000) [inline src/screens/Asset/components/TokenAnalyticsActivityList.tsx:45]:
```suggestion
interface TokenAnalyticsActivityListProps {
```

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  data: PoolActivityData | TredingActivityData;
+  isTreding?: boolean;
+}
+
+const TokenAnalyticsActivityList = ({ data, isTreding }: Props) => {
```
Comment (MayukhSamadder2000) [inline src/screens/Asset/components/TokenAnalyticsActivityList.tsx:50]:
```suggestion
const TokenAnalyticsActivityList = ({ data, isTreding }: TokenAnalyticsActivityListProps) => {
```

Comment (MayukhSamadder2000) [reply src/screens/Asset/components/TokenAnalyticsActivityList.tsx:50]: `isTreding` should be is `isTrading` right?

**Review by MayukhSamadder2000 — CHANGES_REQUESTED** [review]
LOC

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  const fiatValue = fiatTokenValue(amountUSD, ratesPerChain[chain], currency);
+
+  const poolActivityDate = new Date(timestamp * 1000);
+  const currentDate = new Date();
+
+  const formattedTime = getDateDiff(poolActivityDate, currentDate);
+
+  const tokenA = isTreding ? tokensIn[0] : !isEmpty(tokensIn) ? tokensIn[0] : tokensOut[0];
+  const tokenB = isTreding ? tokensOut[0] : !isEmpty(tokensIn) ? tokensIn[1] : tokensOut[1];
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/screens/Asset/components/TokenAnalyticsActivityList.tsx]: I updated it

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+          </Text>
+        </RowContainer>
+      </SubContent>
+    </ItemContainer>
+  );
+};
+
+export default TokenAnalyticsActivityList;
+
+function getDateDiff(startDate, endDate) {
```
[the remark this answers is not in the document]
Comment (lalitkumarrock) [reply src/screens/Asset/components/TokenAnalyticsActivityList.tsx]: It has been moved to the common utils file.

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx (hunk 1 above)
Comment (lalitkumarrock) [reply src/screens/Asset/components/TokenAnalyticsActivityList.tsx]: Updated it.

File: src/screens/Asset/components/TokenAnalyticsActivityList.tsx (hunk 2 above)
Comment (lalitkumarrock) [reply src/screens/Asset/components/TokenAnalyticsActivityList.tsx]: Yes, should be. I updated it

**Review by MayukhSamadder2000 — APPROVED** [review]
LGTM

**Pull request merged**